From 56f4ce37cd52817a5fcba360dc5eec23d81656f0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Florian=20M=C3=BCllner?= <fmuellner@gnome.org>
Date: Fri, 8 Sep 2017 12:13:48 +0200
Subject: [PATCH 2/5] dateMenu: Adjust to GWeather IAPI change

It's not exactly clear what changed - gobject-introspection, gjs - but
the newly added gweather_condition_to_string_full() API no longer works
like it used to. The replacement code does look more idiomatic anyway,
so just fix the code without investigating the reason of the breakage.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=787423
Origin: upstream, 3.27.0, commit:56f4ce37cd52817a5fcba360dc5eec23d81656f0
---
 js/ui/dateMenu.js | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/js/ui/dateMenu.js b/js/ui/dateMenu.js
index 18f2f0e6b..0054f2d02 100644
--- a/js/ui/dateMenu.js
+++ b/js/ui/dateMenu.js
@@ -237,12 +237,12 @@ var WeatherSection = new Lang.Class({
         let options = capitalize ? GWeather.FormatOptions.SENTENCE_CAPITALIZATION
                                  : GWeather.FormatOptions.NO_CAPITALIZATION;
 
-        let [ok, phenom, qualifier] = info.get_value_conditions();
+        let [ok, phenomenon, qualifier] = info.get_value_conditions();
         if (ok)
-            return GWeather.conditions_to_string_full(phenom, qualifier, options);
+            return new GWeather.Conditions({ phenomenon, qualifier}).to_string_full(options);
 
         let [, sky] = info.get_value_sky();
-        return GWeather.sky_to_string_full(sky, options);
+        return GWeather.Sky.to_string_full(sky, options);
     },
 
     _sameSummary: function(info1, info2) {
-- 
2.14.1

